metadata-models/src/main/pegasus/com/linkedin/dataset/DatasetFieldMapping.pdl

```diff
@@ -6,6 +6,7 @@ import com.linkedin.common.DatasetFieldUrn
 /**
  * Representation of mapping between fields in source dataset to the field in destination dataset
  */
+ @deprecated
```

Comment:
is there a pointer to what people should use instead?

Reply:
Alternative pointer is mentioned in the containing DatasetUpstreamLineage record. 
(Can't think of an alternative to this DatasetFieldMapping record by itself. Is that needed?)

The same applies to DatasetFieldUpstream deprecation as well, not sure if that has a clear alternative either by itself.

The fundamental issue seems to be that these "records" are maybe not really standalone definitions; they are defined primarily as "parts" of their "parent" records... not sure if composition vs aggregation can be demarcated in pdl's, and how to do it. For now, it looks like each type resides in its own file regardless of whether it's a composition or aggregation.